### PR TITLE
Gamma 2.0 and migrate to Rust

### PR DESCRIPTION
**Pull request by GAsplund**
Notable changes:
- Super small image size (less than 3 MB)
- Very fast
- Uses Gamma 2.0

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          file: ./Dockerfile
+          push: ${{ github.ref == 'refs/heads/main' }}
+          tags: ghcr.io/cthit/chalmers-ldap-gamma-sync:latest
```
Comment (Oscariremma) [inline .github/workflows/docker.yml]: Maybe we want to use a more through workflow that does linting and tests?
See [hubbit workflows](https://github.com/cthit/hubbit/blob/main/.github/workflows/backend.yml)

**Review by Oscariremma — COMMENTED** [review]
I am ready to improve this if we don't want to fix the workflow right now.

**Comment by GAsplund** [comment]
I could add linting and maybe formatting workflows. I don't intend on creating tests for a program this small however, so I'm not sure what could/should be done there.

**Comment by Oscariremma** [comment]
> I could add linting and maybe formatting workflows. I don't intend on creating tests for a program this small however, so I'm not sure what could/should be done there.

Was probably thinking more on linting and formatting part of the workflows. But it is not critical for merging this.

**Review by Oscariremma — APPROVED** [review]
LGTM

**Pull request merged**